### ISSUE TITLE
Why does changing top_rate and other_rate not affect training accuracy at all?

### ISSUE THREAD

**Issue by chenzhiliang94** [microsoft/LightGBM#2998]
This is not a bug, but rather a really weird observation with respect to the parameters `top_rate` and `other_rate`. AFAIK, they are the key ideas behind GOSS of lightgbm.
## Error message
There is no difference in training accuracy when I change the top_rate and other_rate for the model. Even for really varied changes (0.05 -> 0.5). When I say "no difference", I mean exactly the same accuracy (I checked, there is no error in my code)
## Reproducible examples

I change the parameters like this:
`param["top_rate"] = 0.5 `

`model = lg.train(param, lg_data) `

`param["top_rate"] = 0.05 `

`model = lg.train(param, lg_data)`
![image](https://user-images.githubusercontent.com/25312324/79340158-7bdcad00-7f5c-11ea-9fe1-5d99e41d927a.png)

## Steps to reproduce

1. Just change top_rate in params; observe no change in training/validation accuracy (numerically identical)

**Comment by guolinke** [comment]
What is your learning rate and num_iteration?
return to https://github.com/microsoft/LightGBM/blob/9843506e63c546b989df16e6235261fd38ce247d/src/boosting/goss.hpp#L129, the goss is disabled at the beginning.

**Comment by chenzhiliang94** [comment]
Thank you! This is a direct answer to my question. When I increased number of iteration higher, the accuracy changes.
